### PR TITLE
Add 'override' to features list

### LINKED ISSUE #981: Annotate members that override members from a superclass
It would be great if on a page like the following, members that are defined in this class but override a member from a superclass were annotated as "override", the same way inherited members are annotated as "inherited".

https://domokit.github.io/docs/sky/rendering/RenderSizeObserver-class.html

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #981.

Adds the text 'override' to the features list if the class member is a declared override of the parent.

screen shot 2018-10-15 at 12 07 30 pm

Also handles complex cases involving fields and getter/setter declarations:
screen shot 2018-10-15 at 12 09 27 pm

### CODE REVIEW

**Comment by devoncarew** [comment]
`override` sounds a little odd in this context. `inherited` reads better; perhaps the analog for `override` would be `overridden from parent`? `overrides parent`?

It may additionally make sense to change it from `override` since I don't think this is the same as having an `@override` annotation; one is user added and the other is programmatically calculated.

Should we not show the `inherited` text for nodes that display `override`? It seems like the later implies the former.

I'll review now...

File: lib/src/model.dart [hunk 1]
```diff
@@ -197,6 +213,58 @@ abstract class Inheritable implements ModelElement {
...
   }
+
+  Inheritable get overriddenElement;
+
+  bool _isOverride;
+  bool get isOverride {
+    if (_isOverride == null) {
+      // The canonical version of the enclosing element (not canonicalEnclosingElement,
+      // as that is the element enclosing the canonical version of this element,
+      // two different things.  Defaults to the enclosing element.
```
Comment (devoncarew) [inline lib/src/model.dart]: nit: opening paran not closed

**Comment by jcollins-g** [comment]
@kwalrath Any suggestions on language?  I'm not a huge fan of "override" here.   I agree with Devon that it feels wrong, but I also don't like the alternatives we've thought of so far.

**Comment by kwalrath** [comment]
"overridden" makes more sense to me, since it's parallel to "inherited" and doesn't sound like a command (you must override this).

**Comment by devoncarew** [comment]
Yeah, but in this case it's the parent element which is overridden. Perhaps strict semantics aren't necessary here though, in favor of brevity.

**Comment by jcollins-g** [comment]
@kwalrath Hmmm.  We're really trying to briefly say "This member is inherited from some other class" or "This member overrides a member in some other class" for overrides.

I'm a little concerned about "overridden" because that seems to be common usage for the wrong piece (the part in the other class being, well, overridden).

https://en.wikipedia.org/wiki/Method_overriding
https://en.wikipedia.org/wiki/Method_overriding#/media/File:Method_overriding_in_subclass.svg

"Overriding" is yet another possibility, but doesn't seem that much better than "override" to me.  Despite some discomfort, override does have the advantage of mirroring the Dart annotation for the same thing.

File: lib/src/model.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/model.dart]: fixed

**Pull request merged**